### ISSUE TITLE
Issue 03: coarse に pickability PU loss を追加する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#364]
# Issue 03: coarse に pickability PU loss を追加する

## 背景

Step 2 では、coarse model に trace-level `pickability_logits` を出させる。ただし、manual pick が無い trace は unpickable ではなく unlabeled である。そのため、通常の BCE で `manualなし=0` としてはいけない。

現在の first-break location loss は `meta['fb_idx_view'] > 0` の trace のみで計算されている。この mask を positive として、未ピック trace は PU loss の unlabeled として扱う。

## ゴール

coarse training の criterion を以下にする。

```text
L_total = L_pick_location + λ_pu * L_pickability_nnPU
```

既存設定では `λ_pu = 0` 相当で完全に legacy と同じ。

## 変更対象の候補

- 新規: `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/pickability_loss.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/loss.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/common/loss.py` または coarse 側で wrap
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/train.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/output.py`

### positive / unlabeled mask

batch meta から作る。

```python
manual_pos = batch["meta"]["fb_idx_view"] > 0
trace_valid = batch["meta"].get("trace_valid", torch.ones_like(manual_pos, dtype=bool))
pos_mask = trace_valid & manual_pos
unlabeled_mask = trace_valid & ~manual_pos
```

注意:

- `trace_valid == False` は loss から除外する。negative として扱わない。
- `fb_idx_view <= 0` は unlabeled であり negative ではない。

### nnPU risk

BCEWithLogits を使い、以下を実装する。

```python
loss_pos_as_pos = BCE(logits[pos], ones)
loss_pos_as_neg = BCE(logits[pos], zeros)
loss_unl_as_neg = BCE(logits[unlabeled], zeros)

positive_risk = pi * loss_pos_as_pos
negative_risk = loss_unl_as_neg - pi * loss_pos_as_neg
nnpu_risk = positive_risk + max(negative_risk, 0)
```

`beta` / `gamma` を使う場合は non-negative PU の標準形に近づける。

```python
if negative_risk >= -beta:
    risk = positive_risk + negative_risk
else:
    risk = positive_risk - gamma * negative_risk
```

まずは挙動が安定する実装を優先し、config で `beta=0.0`, `gamma=1.0` を default とする。

### class prior `pi`

`train.pickability_pu.class_prior` が指定されていれば使う。

`None` の場合は、以下のどちらかを実装する。

- A案: config error にして、PU 有効時は `class_prior` 必須にする。
- B案: batch 内 positive ratio を下限付きで使う。

初期実装では A案を推奨する。PU prior が曖昧なまま自動推定すると、selection bias を増幅する可能性がある。

## criterion 統合

`coarse/loss.py::build_criterion()` を coarse 専用の combined criterion にする。

要件:

- `pred` が legacy Tensor の場合、location loss だけを計算する。
- PU enabled なのに `pred` に `pickability_logits` が無い場合は明確なエラー。
- `L_pick_location` はこれまで通り `FbSegKLLossView` を使う。
- `L_pick_location` に渡すのは `get_pick_logits(pred)`。
- PU loss の重みは config から受ける。

`build_train_bundle()` で `build_criterion(typed.train.pickability_pu, ...)` のように typed config を渡す。

## 推奨ログ/メトリクス

training loop は scalar loss しか受け取らないため、この issue ではログ分解は必須にしない。ただし、実装内で debug しやすいように、loss module に `compute_pickability_pu_loss()` を独立関数として切り出す。

## 受け入れ条件

- PU disabled では、既存の location loss と同じ値になる。
- PU enabled で `pickability_logits` が存在すれば、scalar finite loss が返る。
- manual pick あり trace だけが positive として扱われる。
- manual pick なし trace は unlabeled として扱われ、単純 negative BCE にはならない。
- positive trace が batch 内にゼロの場合でも training が落ちない。推奨挙動は PU loss を 0 にして warning なし。
- unlabeled trace が batch 内にゼロの場合でも落ちない。

## テスト

新規テスト候補:

- `tests/test_fbpick_coarse_pickability_pu_loss.py`

テスト内容:

1. legacy Tensor + PU disabled で location loss が動く。
2. dict output + PU enabled で finite loss。
3. PU enabled だが `pickability_logits` が無いと例外。
4. `fb_idx_view <= 0` が negative label として直接 BCE されていないことを、極端な logits で確認する。
5. no positive batch / no unlabeled batch の edge case。

## 非ゴール

- `coarse_usable_logits` の supervised loss は Issue 08。
- OOF error label 生成は Issue 06。
- physics への接続は Issue 04 / Issue 05。
